### PR TITLE
Fitting fixes jvb

### PR DESCRIPTION
**Pull request by joelvbernier**
Adding a new 1-d spectrum fitting module for calibration that uses `lmfit`; this was really precipitated by the need to get the DCS pink beam function working.  Constraints are essential.  That use case is passing basic testing, but need to do several things:

- [x] add unit tests for DCS case (already included) and plain CeO2 (maybe using gaussian and pseudo-Voigt)
- [x] make interface class for generic spectrum fitting with 1 or more peaks and choice of background function
- [x] steal/enhance parameter estimator from `fitpeak`

@saransh13 and @darrencpagan -- I am open to any ideas you have regarding enhancing the initial parameter estimation.  What we currently have is _ok_, but might not be robust if the initial guess is a bit further off.  I am aware that there very well may not be a better generic rubric, but I figured I would plant the seed.

### CODE REVIEW

File: hexrd/fitting/calibration.py [hunk 1]
```diff
@@ -4,14 +4,14 @@
 
 from scipy.optimize import leastsq, least_squares
 
+from tqdm import tqdm
```
Comment (psavery) [inline hexrd/fitting/calibration.py:7]: @joelvbernier Should we add `tqdm` as a dependency in the `setup.py` and conda recipe?

**Review by psavery — APPROVED** [review]
LGTM after we add tqdm as a dependency...

File: hexrd/fitting/calibration.py (hunk 1 above)
Comment (psavery) [reply hexrd/fitting/calibration.py:7]: This was added in #401.

**Comment by joelvbernier** [comment]
Still needs unit tests, but I've tested it with an array of data including DCS.  Also tested the updated "auto" powder calibration on a Pilatus.

**Pull request merged**